### ISSUE TITLE
Multiple version constraints for the same platform in metadata.rb ignored

### ISSUE THREAD

**Issue by ghost** [chef/chef#2457]
Consider the following "platform constraints" in metadata.rb or Cookbook „example“

``` ruby
supports 'redhat', '~> 6.5'
supports 'redhat', '~> 7.0'
supports 'centos', '~> 6.5'
supports 'centos', '~> 7.0'
```

This actually leads to the following merge:

``` ruby
# run_context.cookbook_collection['example'].metadata.platforms

@platforms={"redhat"=>"~> 7.0", "centos"=>"~> 7.0"}
```

It's obvious why this i happening. I'd have been exspecting something like this:

``` ruby
@platforms={"redhat"=> ["~> 6.5", "~> 7.0"], "centos"=> ["~> 6.5", "~> 7.0"]}
```

Unfortunately this would introduce a breaking change.
What do you think?

**Comment by stevendanna** [comment]
@lamont-granquist @yves-vogl We looked at chef/chef-server#31 recently and were a bit confused about why this was moved to the server project.  

The Chef Server has a single purpose in life, fulling requests from chef-client and related API clients.  Changes to it should be driven by actual client use cases.  Looking at the code in the server, it appears we may be able to support multiple constrains without modification (depending on the API signature).  Or, it may require API changes.  However, the question we couldn't answer easily without more input from the client maintainers is "Do we want this feature for knife/chef-client and if so what should it look like?"
